File: packages/ailee-ts/src/domains/video_temporal_provenance/policy.ts
//! Copyright (c) Don Michael Feeney Jr.
//! Licensed under the MIT License.

export interface VideoTemporalPolicy {
  minOverallTrust: number;
  rejectSyntheticTransitions: boolean;
  requiredWatermarkKey?: Uint8Array;
}

export const DEFAULT_VIDEO_TEMPORAL_POLICY: VideoTemporalPolicy = {
  minOverallTrust: 70.0,
  rejectSyntheticTransitions: false,
};

File: ailee/domains/video_temporal_provenance/policy.py
# Copyright (c) Don Michael Feeney Jr.
# Licensed under the MIT License.
"""
Policy structures for Video Temporal Provenance domain.
"""

from dataclasses import dataclass, field
from typing import List

@dataclass
class VideoTemporalPolicy:
    min_overall_trust: float = 70.0
    reject_synthetic_transitions: bool = False
    required_watermark_key: bytes = field(default_factory=bytes)
